uaclient/entitlements/fips.py

```diff
@@ -70,6 +90,25 @@ def static_affordances(self) -> "Tuple[StaticAffordance, ...]":
             ),
         )
 
+        is_xenial_on_blocked_cloud = self._check_for_xenial_cloud_instances()
```

Comment:
It seems strange we are calling this method directly instead of just addding it to static_affordances_list? that lambda in static affordances ensures we will be calling it anyway. We can instead construct the  error_msg = status.MESSAGE_FIPS_BLOCKED_ON_CLOUD.format(cloud_id), lambda: _prevent_fips_on_xenial_cloud_instance

Reply:
Yes, you are right on that. We could do this directly on the existing tuple. I am updating the code